Code/Core/FileIO/FileIO.cpp

```diff
@@ -234,11 +311,26 @@
     // Ensure dest file will be writable if it exists
     FileIO::SetReadOnly( dstFileName, false );
 
-    int dest = open( dstFileName, O_WRONLY | O_CREAT | O_TRUNC, 0644 );
+    int dstFlags = O_WRONLY | O_CREAT | O_TRUNC;
+    const bool noDstFollow = ( linkBehavior != FOLLOW_SRC_AND_DST );
+    if ( noDstFollow )
+    {
+        dstFlags |= O_NOFOLLOW;
+    }
+    int dest = open( dstFileName, dstFlags, 0644 );
     if ( dest < 0 )
     {
-        close( source );
-        return false;
+        if ( ( errno == ELOOP && noDstFollow ) &&
```

Comment:
This probably only comes up when changing link behavior in a CopyFile all, like will happen for me when I go back to the old behavior in my rules